docs/src/cookbooks/cookbook.md

```diff
@@ -231,6 +233,134 @@ class Foo extends RawModule {
 }
 ```
 
+## Bundles
+
+### <a name="aliased-bundle-fields"></a> How do I deal with aliased Bundle fields?
+
+```scala mdoc:invisible:reset
+import chisel3._
+
+class Top[T <: Data](gen: T) extends Module {
+  val in = IO(Input(gen))
+  val out = IO(Output(gen))
+  out := in
+}
+```
+
+Following the `gen` pattern when creating Bundles can result in some opaque error messages:
+
+```scala mdoc
+class AliasedBundle[T <: Data](gen: T) extends Bundle {
+  val foo = gen
+  val bar = gen
+}
+class AlsoAliasedBundle[T <: Data](val gen: T) extends Bundle {
```

Comment:
Can you move this example later and say something like "you'd also get this error in this case..." since you don't talk about this code here it's a bit confusing

Reply:
Sure.